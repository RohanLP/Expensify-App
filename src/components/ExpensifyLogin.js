import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const ExpensifyLogin = ({startLogin}) => (
  <div>
    <h1> Please Log In</h1>
    <input
      type="login"
      placeholder="username"
      
    />

    <input
      type="password"
      placeholder="password"
     
    />

    <button onClick={startLogin}>Log In</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps) (ExpensifyLogin);