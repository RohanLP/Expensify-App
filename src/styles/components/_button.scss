.button {
  background: $deep-blue;
  border: none;
  color: $light-teal;
  display: inline-block;
  font-size: $l-font;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--delete {
  background: $light-teal;
  color: rgb(211, 6, 6);
}

.button--link {
  background: none;
}