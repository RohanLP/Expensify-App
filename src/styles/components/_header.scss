.header {
  background: url('/images/bg.jpg');
}

.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title {
  color: $light-teal;
  text-decoration: none;
  h1 {
    margin: 0
  }
}

