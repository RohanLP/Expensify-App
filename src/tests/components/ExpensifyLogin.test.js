import React from 'react';
import { shallow } from 'enzyme';
import {ExpensifyLogin} from '../../components/ExpensifyLogin';

test('should render ExpensifyLogin correctly', () => {
  const wrapper = shallow(<ExpensifyLogin />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<ExpensifyLogin startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});