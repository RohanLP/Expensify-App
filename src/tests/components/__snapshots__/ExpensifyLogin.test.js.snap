// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensifyLogin correctly 1`] = `
<div>
  <h1>
     Please Log In
  </h1>
  <input
    placeholder="username"
    type="login"
  />
  <input
    placeholder="password"
    type="password"
  />
  <button>
    Log In
  </button>
</div>
`;
